src/ign.cc

```diff
@@ -267,3 +272,87 @@ extern "C" SDFORMAT_VISIBLE int cmdGraph(
 
   return 0;
 }
+
+//////////////////////////////////////////////////
+extern "C" SDFORMAT_VISIBLE int cmdInertialStats(
+    const char *_path)
+{
+  if (!sdf::filesystem::exists(_path))
+  {
+    std::cerr << "Error: File [" << _path << "] does not exist.\n";
+    return -1;
+  }
+
+  sdf::Root root;
+  if (!root.Load(_path).empty()) {
+    return -1;
+  }
+
+  const sdf::Model * model = root.Model();
+  if (!model){
+    return -1;
+  }
+
+  ignition::math::Inertiald totalInertial;
+
+  for (uint64_t i = 0; i < model->LinkCount(); i++) {
+    ignition::math::Pose3d linkPoseRelativeToModel;
+    auto errors = model->LinkByIndex(i)->SemanticPose().
+      Resolve(linkPoseRelativeToModel, "__model__");
+
+    auto currentLinkInertial = model->LinkByIndex(i)->Inertial();
+    currentLinkInertial.SetPose(linkPoseRelativeToModel *
+      currentLinkInertial.Pose());
+    totalInertial += currentLinkInertial;
+  }
+
+  auto totalMass = totalInertial.MassMatrix().Mass();
+  auto xCentreOfMass = totalInertial.Pose().Pos().X();
+  auto yCentreOfMass = totalInertial.Pose().Pos().Y();
+  auto zCentreOfMass = totalInertial.Pose().Pos().Z();
+
+  std::cout << "Inertial statistics for model: " << model->Name() << std::endl;
+  std::cout << "---" << std::endl;
+  std::cout << "Total mass of the model: " << totalMass << std::endl;
+  std::cout << "---" << std::endl;
+
+  std::cout << "Centre of mass in model frame: " << std::endl;
+  std::cout << "X: " << xCentreOfMass << std::endl;
+  std::cout << "Y: " << yCentreOfMass << std::endl;
+  std::cout << "Z: " << zCentreOfMass << std::endl;
+  std::cout << "---" << std::endl;
+
+  std::cout << "Moment of inertia matrix: " << std::endl;
+
+  // Pretty print the MOI matrix
+  std::stringstream ss;
+  ss << totalInertial.Moi();
+
+  std::string s;
+  auto maxLength = 0;
```

Comment:
compiler warning on 336 due to the type of this variable. Use `0u` instead of `0` to fix it (I think)

Reply:
Done